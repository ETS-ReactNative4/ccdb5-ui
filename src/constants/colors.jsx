/* eslint-disable no-inline-comments, camelcase */
module.exports = Object.freeze( {
  BriteCharts: {
    regular: '#20aa3f',
    medium: '#ADDC91',
    light: '#C7E5B3'
  },
  DataLens: {
    0: '#2cb34a',
    1: '#addc91',
    2: '#257675',
    3: '#9ec4c3',
    4: '#0072ce',
    5: '#96c4ed',
    6: '#254b87',
    7: '#9daecc',
    8: '#b4267a',
    9: '#dc9cbf',
    10: '#a2a3a4'
  }
} )
