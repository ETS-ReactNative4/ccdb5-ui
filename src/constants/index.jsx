// Internal triggers
export const PAGE_CHANGED              = 'PAGE_CHANGED'
export const SEARCH_CHANGED            = 'SEARCH_CHANGED'
export const SIZE_CHANGED              = 'SIZE_CHANGED'
export const SORT_CHANGED              = 'SORT_CHANGED'
export const URL_CHANGED               = 'URL_CHANGED'
export const FILTER_CHANGED            = 'FILTER_CHANGED'
export const FILTER_REMOVED            = 'FILTER_REMOVED'
export const FILTER_ALL_REMOVED        = 'FILTER_ALL_REMOVED'

// External Triggers
export const COMPLAINTS_RECEIVED       = 'COMPLAINTS_RECEIVED'
export const COMPLAINTS_FAILED         = 'COMPLAINTS_FAILED'

// Useful constants
export const knownFilters = [
  'company',
  'company_public_response',
  'company_response',
  'consumer_consent_provided',
  'consumer_disputed',
  'issue',
  'product',
  'state',
  'submitted_via',
  'tag',
  'timely',
  'zip_code'
]

export const SLUG_SEPARATOR            = 'â€¢'
