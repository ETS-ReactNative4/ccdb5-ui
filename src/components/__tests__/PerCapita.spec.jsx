import configureMockStore from 'redux-mock-store'
import {
  mapDispatchToProps, mapStateToProps, PerCapita
} from '../RefineBar/PerCapita'
import { Provider } from 'react-redux'
import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import thunk from 'redux-thunk'
import { GEO_NORM_NONE, GEO_NORM_PER1000 } from '../../constants'

function setupEnzyme( normalization = GEO_NORM_NONE ) {
  const props = {
    dataNormalization: normalization,
    enablePer1000: true,
    onDataNormalization: jest.fn()
  }

  const target = shallow(<PerCapita {...props} />);

  return {
    props,
    target
  }
}

function setupSnapshot() {
  const middlewares = [ thunk ]
  const mockStore = configureMockStore( middlewares )
  const store = mockStore( {
    map: {},
    query: {
      company: [ { name: 'foo', value: 123 } ]
    }
  } )

  return renderer.create(
    <Provider store={ store }>
      <PerCapita />
    </Provider>
  )
}

describe( 'component: PerCapita', () => {
  describe( 'initial state', () => {
    it( 'renders without crashing', () => {
      const target = setupSnapshot()
      let tree = target.toJSON()
      expect( tree ).toMatchSnapshot()
    } )

    it('allows the user to trigger Complaints', () => {
      const { target, props } = setupEnzyme()
      const button = target.find('.raw');

      button.simulate('click');
      expect(props.onDataNormalization).toHaveBeenCalled();
    });

    it('allows the user to trigger Per 1000 Complaints', () => {
      const { target, props } = setupEnzyme()
      const button = target.find('.capita');

      expect( button.hasClass( 'capita' ) ).toBeTruthy()
      expect( button.hasClass( 'selected' ) ).toBeFalsy()
      button.simulate('click');
      expect(props.onDataNormalization).toHaveBeenCalled()
    });

    it('changes Per 1000 button class', () => {
      const { target, props } = setupEnzyme( GEO_NORM_PER1000 )
      const button = target.find('.capita')

      expect( button.hasClass( 'capita' ) ).toBeTruthy()
      expect( button.hasClass( 'selected' ) ).toBeTruthy()

      button.simulate('click')
      expect(props.onDataNormalization).toHaveBeenCalled()
    });
  } )

  describe('mapDispatchToProps', () => {
    it('hooks into onDataNormalization', () => {
      const dispatch = jest.fn();
      const ev = {
        target: {
          value: 123
        }
      }
      mapDispatchToProps(dispatch).onDataNormalization( ev );
      expect(dispatch.mock.calls.length).toEqual(1);
    })
  })

  describe( 'mapStateToProps', () => {
    it( 'maps state and props', () => {
      const state = {
        map: {
          dataNormalization: 'foo'
        },
        query: {
          enablePer1000: true
        }
      }
      let actual = mapStateToProps( state )
      expect( actual ).toEqual( {
        dataNormalization: 'foo',
        enablePer1000: true
      } )
    } )
  } )
} )
