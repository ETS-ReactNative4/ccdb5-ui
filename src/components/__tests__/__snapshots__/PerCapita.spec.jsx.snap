// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component: PerCapita initial state renders without crashing 1`] = `
<section
  className="per-capita"
>
  <div
    className="per-capita m-btn-group"
  >
    <p>
      Map shading
    </p>
    <button
      className="a-btn raw deselected"
      onClick={[Function]}
    >
      Complaints
    </button>
    <button
      className="a-btn capita a-btn__disabled"
      onClick={[Function]}
    >
      Complaints per 1,000 
      <span>
        population
      </span>
    </button>
  </div>
</section>
`;
